ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDTFGetSplits.java

```diff
@@ -311,9 +310,10 @@ private PlanFragment createPlanFragment(String query, ApplicationId splitsAppId)
     DriverCleanup driverCleanup = new DriverCleanup(driver, txnManager, splitsAppId.toString());
     boolean needsCleanup = true;
     try {
-      CommandProcessorResponse cpr = driver.compileAndRespond(query, false);
-      if (cpr.getResponseCode() != 0) {
-        throw new HiveException("Failed to compile query: " + cpr.getException());
+      try {
+        driver.compileAndRespond(query, false);
+      } catch (CommandProcessorException e) {
+        throw new HiveException("Failed to compile query: " + e.getException());
```

Comment:
I think we shouldn't call e.getException() 
use it directly - as it contains usefull
and also pass e as the parent exception ; so we will get a full backtrace not just a hiveexception coming from this line